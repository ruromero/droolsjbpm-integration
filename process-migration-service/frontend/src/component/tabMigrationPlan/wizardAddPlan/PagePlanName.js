import React from "react";
import PropTypes from "prop-types";
import {
  FormGroup,
  ControlLabel,
  Form
} from "patternfly-react/dist/js/components/Form";
import FormControl from "patternfly-react/dist/js/components/Form/FormControl";
import { Col } from "patternfly-react/dist/js/components/Grid";
import HelpBlock from "patternfly-react/dist/js/components/Form/HelpBlock";

export default class PagePlanName extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      dirty: false
    };
    this.props.onIsValid(this.isFormValid());
  }
  onDescriptionChange = event => {
    this.props.onFieldChanged(
      "description",
      event.target.value === "" ? null : event.target.value
    );
  };

  onNameChange = event => {
    this.props.onFieldChanged("name", event.target.value);
    this.props.onIsValid(this.isFormValid());
    this.setState({ dirty: true });
  };

  isFormValid = () => {
    return this.props.plan.name !== undefined && this.props.plan.name !== "";
  };

  getValidationState = () => {
    if (this.state.dirty && this.props.plan.name === "") {
      return "error";
    }
    return null;
  };

  render() {
    return (
      <Form horizontal>
        <FormGroup
          controlId="formPlanName"
          validationState={this.getValidationState()}
          className="required"
        >
          <Col componentClass={ControlLabel} sm={3}>
            Name
          </Col>
          <Col sm={9}>
            <FormControl
              type="text"
              value={this.props.plan.name ? this.props.plan.name : ""}
              onChange={this.onNameChange}
            />
            <FormControl.Feedback />
            {this.getValidationState() && (
              <HelpBlock>The name is mandatory</HelpBlock>
            )}
          </Col>
        </FormGroup>
        <FormGroup controlId="formPlanDescription">
          <Col componentClass={ControlLabel} sm={3}>
            Description
          </Col>
          <Col sm={9}>
            <FormControl
              type="text"
              value={
                this.props.plan.description ? this.props.plan.description : ""
              }
              onChange={this.onDescriptionChange}
            />
          </Col>
        </FormGroup>
      </Form>
    );
  }
}

PagePlanName.propTypes = {
  plan: PropTypes.object.isRequired,
  onFieldChanged: PropTypes.func.isRequired,
  onIsValid: PropTypes.func.isRequired
};
