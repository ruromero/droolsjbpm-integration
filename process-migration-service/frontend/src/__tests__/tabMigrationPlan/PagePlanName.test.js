import renderer from "react-test-renderer";
import React from "react";
import PagePlanName from "../../component/tabMigrationPlan/wizardAddPlan/PagePlanName";

test("PagePlanName opens with an empty plan", () => {
  const plan = {};
  const mockFn = jest.fn();
  const tree = renderer
    .create(
      <PagePlanName plan={plan} onFieldChanged={mockFn} onIsValid={mockFn} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("PagePlanName opens using an existing plan", () => {
  const plan = {
    name: "plan name",
    description: "description"
  };
  const mockFn = jest.fn();
  const tree = renderer
    .create(
      <PagePlanName plan={plan} onFieldChanged={mockFn} onIsValid={mockFn} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});