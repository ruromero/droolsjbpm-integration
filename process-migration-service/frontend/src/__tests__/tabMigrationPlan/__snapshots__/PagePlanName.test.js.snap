// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PagePlanName opens using an existing plan 1`] = `
<form
  className="form-horizontal"
>
  <div
    className="required form-group has-feedback"
  >
    <label
      className="col-sm-3 control-label"
      htmlFor="formPlanName"
    >
      Name
    </label>
    <div
      className="col-sm-9"
    >
      <input
        className="form-control"
        id="formPlanName"
        onChange={[Function]}
        type="text"
        value="plan name"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      className="col-sm-3 control-label"
      htmlFor="formPlanDescription"
    >
      Description
    </label>
    <div
      className="col-sm-9"
    >
      <input
        className="form-control"
        id="formPlanDescription"
        onChange={[Function]}
        type="text"
        value="description"
      />
    </div>
  </div>
</form>
`;

exports[`PagePlanName opens with an empty plan 1`] = `
<form
  className="form-horizontal"
>
  <div
    className="required form-group has-feedback"
  >
    <label
      className="col-sm-3 control-label"
      htmlFor="formPlanName"
    >
      Name
    </label>
    <div
      className="col-sm-9"
    >
      <input
        className="form-control"
        id="formPlanName"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      className="col-sm-3 control-label"
      htmlFor="formPlanDescription"
    >
      Description
    </label>
    <div
      className="col-sm-9"
    >
      <input
        className="form-control"
        id="formPlanDescription"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
  </div>
</form>
`;
